main.js

```diff
@@ -0,0 +1,239 @@
+let counter = document.getElementById("counter")
+let minutes = document.getElementById("min")
+let seconds = document.getElementById("sec")
+seconds.textContent = '00'
+minutes.textContent = '00'
+let radioAI = document.getElementById("PvAI")
+let radioPvP = document.getElementById("PvP")
+let radios = document.getElementsByClassName('radio')
+let gameAI
+let squareArr = [square0, square1, square2, square3, square4, square5, square6, square7, square8]
+let winArrays = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
+const board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
+let player1 = {
+    symbol: 'X',
+    name: '',
+    moves: []
+}
+let player2 = {
+    symbol: 'O',
+    name: '',
+    moves: []
+}
+let currentPlayer = player1
+////////////////////////////////////////functions////////////////////////////////////
+function changePlayer() {
+    if (currentPlayer === player1) {
+        currentPlayer = player2
+    } else currentPlayer = player1
+}
+//compares 2 arrays; used in checkWin
+function compareArray(arr1, arr2) {
+    for (let i = 0; i < arr2.length; i++) {
+        if (!arr1.includes(arr2[i])) {
+            return false
+        }
+    }
+    return true
+}
+//compares player array to set of win arrays
+function checkWin(array, nestArr) {
+    for (win of nestArr) {
+        if (compareArray(array, win)) {
+            return true
+        }
+    } return false
+}
+// returns previously discovered win array (for highlighting)
+function defineWin(array, nestArr) {
+    for (win of nestArr) {
+        if (compareArray(array, win)) {
+            return win
+        }
+    } return false
+}
+// highlights win array 
+function drawLine(arr) {
+    for (square of arr) {
+        squareArr[square].style.color = 'red'
+    }
+}
+//creates a instance of possible moves based on both player's moves
+function buildBoard(p1Arr, p2Arr) {
+    let arr = []
+    for (let move of board) {
+        if (!p1Arr.includes(move) && !p2Arr.includes(move)) {
+            arr.push(move)
+        }
+    }
+    return arr
+}
+//begins counter interval
+function startCounter() {
+    counting = setInterval(() => {
+        //increments every given time (in milliseconds)
+        seconds.textContent = +seconds.textContent + 1
+        if (+seconds.textContent < 10 && +seconds.textContent > 0) {
+            seconds.textContent = '0' + seconds.textContent
+        }
+        if (+seconds.textContent > 59) {
+            minutes.textContent = +minutes.textContent + Math.floor(+seconds.textContent / 60)
+            if (60 % +seconds.textContent == 0) {
+                seconds.textContent == '00'
+            }
+            if (+minutes.textContent < 10) {
+                minutes.textContent = '0' + +minutes.textContent
+            }
+            seconds.textContent = +seconds.textContent - 60
+            if (seconds.textContent == 0) {
+                seconds.textContent = '00'
+            }
+        }
+    }, 100)
```

Comment:
Your timer function works great! You just have your interval set to 100 milliseconds, so it's counting a bit fast!